app/Http/Controllers/Bible/BibleFilesetsDownloadController.php

```diff
@@ -0,0 +1,190 @@
+        $limit = max($limit, 5000);
+
+        $cache_params = $this->removeSpaceFromCacheParameters([$fileset_id, $book_id, $chapter]);
+
+        $fileset_chapters = cacheRemember(
+            $cache_key,
+            $cache_params,
+            now()->addHours(12),
+            function () use ($fileset_id, $book_id, $chapter, $type, $limit) {
+                $fileset_from_id = BibleFileset::where('id', $fileset_id)->first();
+                if (!$fileset_from_id) {
+                    return $this->setStatusCode(404)->replyWithError(
+                        trans('api.bible_fileset_errors_404')
+                    );
+                }
+
+                $fileset_type = $fileset_from_id['set_type_code'];
+                // fixes data issue where text filesets use the same filesetID
+                $fileset_type = $this->getCorrectFilesetType($fileset_type, $type);
+                $fileset = BibleFileset::with('bible')
+                    ->uniqueFileset($fileset_id, $fileset_type)
+                    ->first();
+                if (!$fileset) {
+                    return $this->setStatusCode(404)->replyWithError(
+                        trans('api.bible_fileset_errors_404')
+                    );
+                }
+
+                $bulk_access_control = $this->allowedForDownload($fileset);
+  
+                if (isset($bulk_access_control->original['error'])) {
+                    return $bulk_access_control;
+                }
+
+                $asset_id = $fileset->asset_id;
+                $bible = optional($fileset->bible)->first();
+
+                $book = $book_id
+                    ? Book::where('id', $book_id)
+                        ->orWhere('id_osis', $book_id)
+                        ->orWhere('id_usfx', $book_id)
+                        ->first()
+                    : null;
+
+                if ($fileset_type === 'text_plain') {
+                    return $this->showTextFilesetChapter(
+                        $limit,
+                        $bible,
+                        $fileset,
+                        $book,
+                        $chapter
+                    );
+                } else {
+                    return $this->showAudioVideoFilesets(
+                        $limit,
+                        $bible,
+                        $fileset,
+                        $asset_id,
+                        $fileset_type,
+                        $book,
+                        $chapter
+                    );
+                }
+            }
+        );
+
+        return $this->reply($fileset_chapters, [], '');
+    }
+
+    /**
+     *
+     * @OA\Get(
+     *     path="download/list",
+     *     tags={"Bibles"},
+     *     summary="List of filesets which can be downloaded for this API key",
+     *     description="List of filesets which can be downloaded for this API key",
+     *     operationId="v4_internal_bible_filesets.list",
```

Comment:
@vichugofsl please rename this operationId to v4_download_list

Reply:
@bradflood It has been removed